// @coinbase/onchainkit/dist/lib/core/getFrameMessage.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const mock_1 = require("./mock");
const getFrameMessage_1 = require("./getFrameMessage");
const neynarUserFunctions_1 = require("../utils/neynar/user/neynarUserFunctions");
jest.mock('../utils/neynar/user/neynarUserFunctions', () => {
    return {
        neynarBulkUserLookup: jest.fn(),
    };
});
jest.mock('@farcaster/hub-nodejs', () => {
    return {
        getSSLHubRpcClient: jest.fn().mockReturnValue({
            validateMessage: jest.fn(),
        }),
        Message: {
            decode: jest.fn(),
        },
    };
});
describe('getFrameValidatedMessage', () => {
    it('should return undefined if the message is invalid', async () => {
        const fid = 1234;
        const addresses = ['0xaddr1'];
        const { validateMock } = (0, mock_1.mockNeynarResponse)(fid, addresses, neynarUserFunctions_1.neynarBulkUserLookup);
        validateMock.mockClear();
        validateMock.mockResolvedValue({
            isOk: () => {
                return false;
            },
        });
        const result = await (0, getFrameMessage_1.getFrameMessage)({
            trustedData: { messageBytes: 'invalid' },
        });
        expect(result?.isValid).toEqual(false);
    });
    it('should return the message if the message is valid', async () => {
        const fid = 1234;
        const addresses = ['0xaddr1'];
        (0, mock_1.mockNeynarResponse)(fid, addresses, neynarUserFunctions_1.neynarBulkUserLookup);
        const fakeFrameData = {
            trustedData: {},
        };
        const result = await (0, getFrameMessage_1.getFrameMessage)(fakeFrameData);
        expect(result?.message?.fid).toEqual(fid);
    });
});
