// @coinbase/onchainkit/dist/lib/core/mock.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.mockNeynarResponse = exports.buildFarcasterResponse = void 0;
function buildFarcasterResponse(fid) {
    return {
        isOk: () => {
            return true;
        },
        value: {
            valid: true,
            message: {
                data: {
                    fid: fid,
                },
            },
        },
    };
}
exports.buildFarcasterResponse = buildFarcasterResponse;
function mockNeynarResponse(fid, addresses, lookupMock) {
    const neynarResponse = {
        users: [
            {
                verifications: addresses,
            },
        ],
    };
    lookupMock.mockResolvedValue(neynarResponse);
    const getSSLHubRpcClientMock = require('@farcaster/hub-nodejs').getSSLHubRpcClient;
    const validateMock = getSSLHubRpcClientMock().validateMessage;
    // Mock the response from the Farcaster hub
    validateMock.mockResolvedValue({
        isOk: () => true,
        value: { valid: true, message: { fid } },
    });
    validateMock.mockResolvedValue(buildFarcasterResponse(fid));
    return {
        validateMock,
    };
}
exports.mockNeynarResponse = mockNeynarResponse;
